### ISSUE TITLE
annotation loss with big object

### ISSUE THREAD

**Issue by jpointchoi** [ultralytics/ultralytics#14315]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

Noticed a intersting behaviour during segmentation training
Realtively big object annotation is lost with training batch as shown in the picture. Wondering if this happens with scaling up built-in augmentation behaves unexpectedly. Please advise

Thanks
j

![annotation_missed_with_scaling_up](https://github.com/ultralytics/ultralytics/assets/140245500/3cfbf7b7-127f-4dd8-8e1f-f73f156eaa3c)

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
Hi @jpointchoi,

Thank you for bringing this to our attention! The issue you're describing with the loss of annotations for relatively large objects during segmentation training could indeed be related to the scaling and augmentation processes.

To better assist you, could you please provide a minimal reproducible example of your code? This will help us diagnose the issue more effectively. You can find guidelines on how to create a minimal reproducible example [here](https://docs.ultralytics.com/help/minimum_reproducible_example).

Additionally, please ensure that you are using the latest versions of all relevant packages, including the Ultralytics YOLOv8 library. Sometimes, issues are resolved in newer releases, and updating might solve the problem.

Looking forward to your response so we can help you resolve this! 😊

**Comment by jpointchoi** [comment]
> ### Search before asking
> * [x]  I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.
> 
> ### YOLOv8 Component
> _No response_
> 
> ### Bug
> Noticed a intersting behaviour during segmentation training Realtively big object annotation is lost with training batch as shown in the picture. Wondering if this happens with scaling up built-in augmentation behaves unexpectedly. Please advise
> 
> Thanks j
> 
> ![annotation_missed_with_scaling_up](https://private-user-images.githubusercontent.com/140245500/347267440-3cfbf7b7-127f-4dd8-8e1f-f73f156eaa3c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjExMDQyOTAsIm5iZiI6MTcyMTEwMzk5MCwicGF0aCI6Ii8xNDAyNDU1MDAvMzQ3MjY3NDQwLTNjZmJmN2I3LTEyN2YtNGRkOC04ZTFmLWY3M2YxNTZlYWEzYy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzE2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcxNlQwNDI2MzBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03YmIyNzBmNzNhMTgyMjNiNzhlODYwY2IwN2NiNTZiYTgzZjMyNGI2MDkxYjJjNmZkYzFiOWNhNDk3NzJkNTcxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.diaA2VzsU7z1CnU2F1uB5mRKGj55fupjv2g89loPUqU)
> 
> ### Environment
> _No response_
> 
> ### Minimal Reproducible Example
> _No response_
> 
> ### Additional
> _No response_
> 
> ### Are you willing to submit a PR?
> * [ ]  Yes I'd like to help by submitting a PR!



> Hi @jpointchoi,
> 
> Thank you for bringing this to our attention! The issue you're describing with the loss of annotations for relatively large objects during segmentation training could indeed be related to the scaling and augmentation processes.
> 
> To better assist you, could you please provide a minimal reproducible example of your code? This will help us diagnose the issue more effectively. You can find guidelines on how to create a minimal reproducible example [here](https://docs.ultralytics.com/help/minimum_reproducible_example).
> 
> Additionally, please ensure that you are using the latest versions of all relevant packages, including the Ultralytics YOLOv8 library. Sometimes, issues are resolved in newer releases, and updating might solve the problem.
> 
> Looking forward to your response so we can help you resolve this! 😊


Hi Glenn, 
I was able to replicate this issue. You can try below code, which deosn't generate segmenation anotation when the scaling is 0.5.

import os.path
import cv2
import numpy as np

from pathlib import Path
def generate_mask_labels(dataset):
    train_images_dir=os.path.join(dataset, "train/images")
    train_labels_dir=os.path.join(dataset, "train/labels")
    val_images_dir=os.path.join(dataset, "val/images")
    val_labels_dir=os.path.join(dataset, "val/labels")
    Path(train_images_dir).mkdir(parents=True, exist_ok=True)
    Path(val_images_dir).mkdir(parents=True, exist_ok=True)

    # 이미지 생성 (예: 흰색 배경)
    width, height = 640, 640
    image = np.ones((height, width, 3), dtype=np.uint8) * 255
    # 박스 좌표 (왼쪽 상단 및 오른쪽 하단 좌표)
    x1, y1, x2, y2 = 10, 10, 630, 630
    # 마스크 생성 (박스 내부는 흰색, 나머지는 검정색)
    mask = np.zeros((height, width), dtype=np.uint8)
    mask[y1:y2, x1:x2] = 255
    cv2.imwrite(os.path.join(train_images_dir,'mask.png'), mask)
    cv2.imwrite(os.path.join(val_images_dir, 'mask.png'), mask)
    cv2.imshow('Mask', mask)
    cv2.waitKey(0)
    cv2.destroyAllWindows()
    # 외곽선 찾기
    contours, _ = cv2.findContours(mask, cv2.RETR_EXTERNAL, cv2.CHAIN_APPROX_SIMPLE)
    # 이미지 크기 얻기
    height, width = mask.shape
    # 다각형 데이터 저장 리스트
    polygons = []
    # 각 외곽선에 대해
    for contour in contours:
        # 다각형 근사화
        epsilon = 0.01 * cv2.arcLength(contour, True)
        approx = cv2.approxPolyDP(contour, epsilon, True)

        # 다각형 포인트를 YOLO 형식으로 변환
        polygon = []
        for point in approx:
            x, y = point[0]
            norm_x = x / width
            norm_y = y / height
            polygon.append((norm_x, norm_y))

        polygons.append(polygon)
    # 폴리곤 출력
    for i, polygon in enumerate(polygons):
        print(f"Polygon {i + 1}:")
        for point in polygon:
            print(point)
    # 클래스 ID (예: 0)
    class_id = 0
    # YOLOv8 라벨 형식으로 저장

    Path(train_labels_dir).mkdir(parents=True, exist_ok=True)
    Path(val_labels_dir).mkdir(parents=True, exist_ok=True)

    with open(os.path.join(train_labels_dir,'mask.txt'), 'w') as f:
        for polygon in polygons:
            label = f"{class_id}"
            for x, y in polygon:
                label += f" {x} {y}"
            f.write(label + '\n')
    with open(os.path.join(val_labels_dir, 'mask.txt'), 'w') as f:
        for polygon in polygons:
            label = f"{class_id}"
            for x, y in polygon:
                label += f" {x} {y}"
            f.write(label + '\n')
    print("YOLOv8 폴리곤 라벨이 'labels_polygon.txt' 파일에 저장되었습니다.")

import yaml
def generate_yaml(path, train, val, nc, names, data_yaml):
    yaml_text = "path: " + path + " \n"
    yaml_text += "train: " + train + " \n"
    yaml_text += "val: " + val + " \n"
    yaml_text += "nc: " + str(nc) + " \n"
    yaml_text += "names: " + str(names)

    # names_yaml = """
    # - 'path'
    # - 'justin'
    # - 'mary-kate'
    # """

    names = yaml.safe_load(yaml_text)

    with open(data_yaml, 'w') as file:
        yaml.dump(names, file)

    # print(open('names.yaml').read())

from ultralytics import YOLO
import wandb
if __name__ == '__main__':
    project="scaling_augmentation_test"
    current_directory = os.getcwd()
    dataset_base=os.path.join(current_directory,project)
    generate_mask_labels(dataset_base)

    num_of_class = 1
    label_list = ['mask']
    data_yaml = project + ".yaml"
    train_image_dir = 'train/images/' 
    val_image_dir = 'val/images/'

    generate_yaml(os.path.join(current_directory,project), train_image_dir, val_image_dir, num_of_class, label_list, data_yaml)

    wandb.init(project=project, name="", job_type="train")
    # Load a model
    model = YOLO('yolov8x-seg')  # load a pretrained model (recommended for training)
    results = model.train(project=project, name="training_result", data=data_yaml, batch=1, epochs=10,
                          imgsz=640, device=7, plots=True, exist_ok=True, val=True, scale=0.5).

**Comment by jpointchoi** [comment]
It looks like if the scaled up object exceeds 640x640, then deletete the polygon instead of using 640x640 feature...

**Comment by pderrenger** [comment]
Thanks for sharing your observations! It does seem like the scaling might be causing the issue. Could you please check if this behavior persists with the latest version of the Ultralytics YOLOv8 library? Sometimes updates include fixes for such issues. If the problem continues, let us know, and we'll look into it further.

**Comment by jpointchoi** [comment]
> Thanks for sharing your observations! It does seem like the scaling might be causing the issue. Could you please check if this behavior persists with the latest version of the Ultralytics YOLOv8 library? Sometimes updates include fixes for such issues. If the problem continues, let us know, and we'll look into it further.

just verified that the issue still persist with the version of 
"Successfully installed ultralytics-8.2.59"

**Comment by pderrenger** [comment]
Thank you for verifying with the latest version. Given that the issue persists, it appears to be a bug related to the scaling augmentation. We'll investigate this further. In the meantime, you might consider adjusting the augmentation parameters or temporarily disabling scaling to avoid this issue. We appreciate your patience and will update you once we have more information.
